streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorStateManagerTest.java

```diff
@@ -362,6 +368,9 @@ public void testFlushAndClose() throws IOException {
         assertTrue(nonPersistentStore.closed);
         assertTrue(checkpointFile.exists());
 
+        // make sure that flush is called in the proper order
+        assertThat(persistentStore.getLastFlushCount(), Matchers.lessThan(nonPersistentStore.getLastFlushCount()));
```

Comment:
Here's the regression test for my code change. Previously, we would flush the stores in registration order, but now we do it in topological order (defined on L346).